lightpath/path.py

```diff
@@ -171,125 +185,114 @@ def __init__(self, *devices, name=None):
         :attr:`.impediment`
+
+        Returns
+        -------
+        List[Device]
+            list of blockoing devices
         """
         # Cache important prior devices
-        prior = None
-        last_branches = list()
+        prev_device = None
+        prev_status = None
         block = list()
         for device in self.path:
-            # If we have switched beamlines
-            if prior and device.md.beamline != prior.md.beamline:
-                # Find improperly configured optics
-                for optic in last_branches:
-                    # If this optic is responsible for delivering beam
-                    # to this hutch and it is not configured to do so.
-                    # Mark it as blocking
-                    if device.md.beamline in optic.branches:
-                        if device.md.beamline not in optic.destination:
-                            block.append(optic)
-                    # Otherwise ensure it is removed from the beamline
-                    elif optic.md.beamline not in optic.destination:
-                        block.append(optic)
-                # Clear optics that have been evaluated
-                last_branches.clear()
-
-            # If our last device was an optic, make sure it wasn't required
-            # to continue along this beampath
-            elif (prior in last_branches
-                    and device.md.beamline in prior.branches
-                    and device.md.beamline not in prior.destination):
-                block.append(last_branches.pop(-1))
-
-            # Find branching devices and store
-            # They will be marked as blocking by downstream devices
-            dev_state = find_device_state(device)
-            if device in self.branches:
-                last_branches.append(device)
-            # Find inserted devices
-            elif dev_state == DeviceState.Inserted:
+            curr_state, curr_status = find_device_state(device)
+
+            # short circuit if statuses are in error
+            if curr_state in (DeviceState.Error, DeviceState.Unknown):
+                block.append(device)
+
+            # check to make sure input and output branches match
+            # e.g. mirror not pointing to current device
+            elif (prev_device and prev_status and
+                  prev_status.output_branch not in device.input_branches):
+                block.append(prev_device)
+            # check inserted
+            elif curr_state is DeviceState.Inserted:
                 # Ignore devices with low enough transmssion
-                trans = getattr(device, 'transmission', 1)
-                if trans < self.minimum_transmission:
+                if curr_status.transmission < self.minimum_transmission:
                     block.append(device)
             # Find unknown and faulted devices
-            elif dev_state != DeviceState.Removed:
+            elif curr_state is not DeviceState.Removed:
```

Comment:
This should be caught by the above condition, no?  I suppose we could make this more explicitly an `else` statement

Reply:
I was quick to jump on this - I wanted to make sure I saw every `DeviceState` referenced once and got distracted with the actual logic. 

Just in case, there should be an `else` clause, too, that is considered the same as "errored".